File: src/main/java/com/example/today_i_baekjoon/service/UserService.java
package com.example.today_i_baekjoon.service;

import com.example.today_i_baekjoon.domain.User;
import com.example.today_i_baekjoon.repository.UserRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
@RequiredArgsConstructor
public class UserService {
    
    private final UserRepository userRepository;

    public Optional<User> findUserByUsername(String username) {
        return userRepository.findByUsername(username);
    }

    public User createUser(String username) {
        return userRepository.save(new User(username));
    }
    
    public List<User> findAll() {
        return userRepository.findAll();
    }
}

File: back/src/main/kotlin/com/gidskql6671/today_i_baekjoon/service/UserService.kt
package com.gidskql6671.today_i_baekjoon.service

import com.gidskql6671.today_i_baekjoon.domain.User
import com.gidskql6671.today_i_baekjoon.repository.UserRepository
import org.springframework.stereotype.Service
import kotlin.jvm.optionals.getOrNull

@Service
class UserService(
    private val userRepository: UserRepository
) {

    fun findUser(userId: Long): User? {
        return userRepository.findById(userId).getOrNull()
    }

    fun findUserByUsername(username: String): User? {
        return userRepository.findByUsername(username)
    }

    fun createUser(username: String): User {
        return userRepository.save(User(username = username))
    }

    fun findAll(): List<User> {
        return userRepository.findAll()
    }
}